genesyscloud/recording_media_retention_policy/resource_genesyscloud_recording_media_retention_policy_test.go

```diff
@@ -1372,7 +1480,120 @@ func TestAccResourceMediaRetentionPolicyBasic(t *testing.T) {
 					resource.TestCheckResourceAttr("genesyscloud_recording_media_retention_policy."+policyResourceLabel1, "media_policies.0.email_policy.0.conditions.0.date_ranges.0", fmt.Sprint(mediaRetentionEmailPolicy.MediaPolicies.EmailPolicy.Conditions.DateRanges[0])),
 				),
 			},
+			{
 
+				Config: routingEmailDomain.GenerateRoutingEmailDomainResource(
+					domainResourceLabel,
+					domainId,
+					util.FalseValue, // Subdomain
+					util.NullValue,
+				) + routingQueue.GenerateRoutingQueueResourceBasic(queueResourceLabel1, queueName, "") +
+					authRole.GenerateAuthRoleResource(
+						roleResourceLabel1,
+						roleName1,
+						roleDesc1,
+						authRole.GenerateRolePermissions(permissions...),
+						authRole.GenerateRolePermPolicy(qualityDomain, evaluationEntityType, strconv.Quote(editAction)),
+						authRole.GenerateRolePermPolicy(qualityDomain, calibrationEntityType, strconv.Quote(addAction)),
+					) +
+					userRoles.GenerateUserRoles(
+						userRoleResourceLabel1,
+						userResourceLabel1,
+						GenerateResourceRoles("genesyscloud_auth_role."+roleResourceLabel1+".id"),
+					) +
+					generateUserWithCustomAttrs(userResourceLabel1, userEmail, userName) +
+					gcloud.GenerateEvaluationFormResource(evaluationFormResourceLabel1, &evaluationFormResourceBody) +
+					gcloud.GenerateSurveyFormResource(surveyFormResourceLabel1, &surveyFormResourceBody) +
+					integration.GenerateIntegrationResource(integrationResourceLabel1, strconv.Quote(integrationIntendedState), strconv.Quote(integrationType), "") +
+					routinglanguage.GenerateRoutingLanguageResource(languageResourceLabel1, languageName) +
+					authDivision.GenerateAuthDivisionBasic(divResourceLabel, divName) +
+					routingWrapupcode.GenerateRoutingWrapupcodeResource(wrapupCodeResourceLabel1, wrapupCodeName, "genesyscloud_auth_division."+divResourceLabel+".id", description) +
+					architect_flow.GenerateFlowResource(
+						flowResourceLabel1,
+						filePath1,
+						"",
+						false,
+						util.GenerateSubstitutionsMap(map[string]string{
+							"flow_name":            flowName,
+							"default_language":     "en-us",
+							"greeting":             "Archy says hi!!!",
+							"menu_disconnect_name": "Disconnect",
+						}),
+					) +
+					team.GenerateTeamResource(
+						teamIdResourceLabel1,
+						teamIdName1,
+						"genesyscloud_auth_division."+divResourceLabel+".id",
+						teamDescription1,
+					) +
+					generateMediaRetentionPolicyResource(policyResourceLabel1, &mediaRetentionMessagePolicy2),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
The function `ValidateResourceAttributeInArray` is useful here for testing that the id of another resource is present in a list